### ISSUE TITLE
ArgumentNullException when building WebGL

### ISSUE THREAD

**Issue by tiennvsolar** [continuation-team/unity-ton-connect#4]
I'm using unity to build webgl, but when connecting to tonkeeper I get this error. How can I fix it?
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/c6e88f25-323e-4498-8298-0ed2d5575ec9)

- I'm using TC_minimal build, .net standard 2.1, compression format disable
- When connecting to tonkeeper, i got qr code with parameter r={}. I fix that url and got above error

**Comment by purpleguy99** [comment]
have you passed your own qr code with own qr link?

**Comment by tiennvsolar** [comment]
The url i got from tonConnectHandler.tonConnect.Connect(walletConfig):
https://app.tonkeeper.com/ton-connect?v=2&id=2DC994B62155EF1E67EB3D3F9AE019584828CE17FF8B130C2029502102C23A06&r={}
if i use it, it causes error string pattern when i scan QR code.
Then i fix it with manifest url, login success in tonkeeper but cause ArgumentNullException in web
https://app.tonkeeper.com/ton-connect?v=2&id=2DC994B62155EF1E67EB3D3F9AE019584828CE17FF8B130C2029502102C23A06&r=%7b%22manifestUrl%22%3a+%22https%3a%2f%2fpub-5113f1001b074287b5d93bf0917f2da0.r2.dev%2fmainfest.json%22%2c+%22items%22%3a+%5b%7b%22name%22%3a+%22ton_addr%22%7d%5d%7d

**Comment by purpleguy99** [comment]
send me all info what you did to fix it, what you recive on both cases

**Comment by purpleguy99** [comment]
try to clone unity-ton-connect repo and open it like unity project and make web build. After that check it how it works

**Comment by tiennvsolar** [comment]
This is my code for generating QR
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/c5f4b6fd-0cf7-43ff-a8df-24d9cb055454)
If i use defaultUrlNotFix, it will cause error string pattern. The demo project also has the same problem.
Both work very well in unity editor. But error when build to WebGL
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/c95ee4cd-cd97-42d6-9d60-e8b7cd05abc7)

**Comment by purpleguy99** [comment]
try to clone unity-ton-connect repo and open it like unity project and make web build. After that check it how it works

**Comment by purpleguy99** [comment]
it no need any corrections in qr working

**Comment by tiennvsolar** [comment]
The demo project which i talk about is unity-ton-connect. When i build to webgl, If i use web tonkeeper from telegram, i cause error string pattern. If i use tonkeeper app from applestore, it'll cause ArgumentNullException. Here is unity-ton-connect repo build
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/10af6ae3-b38a-453b-8900-c86b01da6cc8)

**Comment by tiennvsolar** [comment]
It works very well in unity editor, only error when i build to webgl

**Comment by purpleguy99** [comment]
What manifset you are using, show me settings and also build settings

**Comment by tiennvsolar** [comment]
This is my manifest: https://pub-5113f1001b074287b5d93bf0917f2da0.r2.dev/mainfest.json
Build settings:
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/72cd1c20-956b-4bfd-9b0e-484923b20069)
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/70199a28-abc7-4f81-a86c-964412a92c64)
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/71e2e808-3a86-4f8d-81f2-1bd3bf01b409)
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/d1aacc1d-d2c0-4b2a-b0dd-fe29a414d653)
![image](https://github.com/continuation-team/unity-ton-connect/assets/74697768/74e358de-6f32-47e5-a1cd-c80040cb4490)

**Comment by purpleguy99** [comment]
Run all like in repo

**Comment by purpleguy99** [comment]
Because in repo different config

**Comment by tiennvsolar** [comment]
Yeah. I have just clone and set default config. It works but only with tonkeeper application in appstore. Both mainet and testnet are not woking, stuck at qr scan screen

**Comment by purpleguy99** [comment]
ok i will test it with your manifest

**Comment by purpleguy99** [comment]
are you using build and run option?

**Comment by tiennvsolar** [comment]
Yes, I'm just clone and click build and run

**Comment by purpleguy99** [comment]
ok let me check it 2 min

**Comment by purpleguy99** [comment]
because it takes a lot 
![изображение](https://github.com/continuation-team/unity-ton-connect/assets/89442150/812fd3f1-552a-4a18-892f-384091041910)

**Comment by purpleguy99** [comment]
so tell me, with what wallets it working and with what its not working?

**Comment by purpleguy99** [comment]
i know whats going on

**Comment by tiennvsolar** [comment]
Tonkeeper app download from apple store is working well but Tonkeeper web from telegram not working

**Comment by purpleguy99** [comment]
so ok,, can you please open build settings
![изображение](https://github.com/continuation-team/unity-ton-connect/assets/89442150/ea290f70-e6bb-4370-9cc3-5e5aa7b4a813)
and disable code stripping like on screen

**Comment by tiennvsolar** [comment]
Thank you but there is no difference from before. It's still stuck at scan qr code screen

**Comment by purpleguy99** [comment]
> Thank you but there is no difference from before. It's still stuck at scan qr code screen

shot the video with all actions you do and what you getting

**Comment by tiennvsolar** [comment]
https://github.com/continuation-team/unity-ton-connect/assets/74697768/9a6e4689-4ef1-495d-a554-585d2a1a5aa8
Here, if i use telegram tonkeeper, it'll be stuck at this screen

**Comment by purpleguy99** [comment]
is it only with telegram tonkeeper?

**Comment by tiennvsolar** [comment]
I just use two tonkeeper
Telegram tonkeeper is not working
This tonkeeper works well: https://apps.apple.com/us/app/tonkeeper-ton-wallet/id1587742107

**Comment by purpleguy99** [comment]
> I just use two tonkeeper Telegram tonkeeper is not working This tonkeeper works well: https://apps.apple.com/us/app/tonkeeper-ton-wallet/id1587742107

ok i will check it in few days, dont use web at the moment

**Comment by DavidNightinga1e** [comment]
I've had same isssue, fixed by disabling "Strip Engine Code" and setting "Managed Stripping Level" to "Minimal"
![image](https://github.com/continuation-team/unity-ton-connect/assets/43321460/8d38f010-fdea-4d1a-9c0f-60a8c29a80ce)
